File: harmony-core/src/Core/Time.cpp
#include <optick.h>
#include "Core/Time.h"

double harmony::Time::GetFrameTime() {
    OPTICK_EVENT();
    return p_FrameTime * p_TimeScale;
}

double harmony::Time::GetFrameTimeUnscaled() {
    OPTICK_EVENT();
    return p_FrameTime;
}

double harmony::Time::GetTimeScale() {
    OPTICK_EVENT();
    return p_FrameTime;
}

void harmony::Time::SetTimeScale(double timeScale) {
    OPTICK_EVENT();
    p_TimeScale = timeScale;
}

File: projects/MonoTest/C#/HarmonyMono/Time.cs

using System.Runtime.CompilerServices;

namespace Harmony
{
    public static class Time
    {
        [MethodImpl(MethodImplOptions.InternalCall)]
        public extern static double GetFrameTime();
        
        [MethodImpl(MethodImplOptions.InternalCall)]
        public extern static double GetFrameTimeUnscaled();

        [MethodImpl(MethodImplOptions.InternalCall)]
        public extern static double GetTimeScale();

        [MethodImpl(MethodImplOptions.InternalCall)]
        public extern static double SetTimeScale(float newTimeScale);
    }
}
